order-book/src/test/java/com/market/banica/order/book/model/ItemMarketTest.java

```diff
@@ -32,6 +40,73 @@ public void setUp() {
         allItems.put(RICE_ITEM_NAME, new TreeSet<>());
 
         ReflectionTestUtils.setField(itemMarket, ALL_ITEMS_FIELD, allItems);
+        ReflectionTestUtils.setField(itemMarket, PRODUCTS_QUANTITY_FIELD, productsQuantity);
+    }
+
+    @Test
+    public void updateItemUpdatesProductQuantityAndAllItemsMaps() {
+        //Arrange
+        itemMarket.addTrackedItem(CHEESE_ITEM_NAME);
+        TickResponse cheese = TickResponse.newBuilder().setGoodName(CHEESE_ITEM_NAME).setQuantity(2).setPrice(2.6).setOrigin(Origin.ASIA).build();
+        Aurora.AuroraResponse build = Aurora.AuroraResponse.newBuilder().setMessage(Any.pack(cheese)).build();
+
+        TickResponse cheese2 = TickResponse.newBuilder().setGoodName(CHEESE_ITEM_NAME).setQuantity(2).setPrice(2.6).setOrigin(Origin.ASIA).build();
+        Aurora.AuroraResponse build2 = Aurora.AuroraResponse.newBuilder().setMessage(Any.pack(cheese2)).build();
+
+        TickResponse cheese3 = TickResponse.newBuilder().setGoodName(CHEESE_ITEM_NAME).setQuantity(2).setPrice(2.6).setOrigin(Origin.AMERICA).build();
+        Aurora.AuroraResponse build3 = Aurora.AuroraResponse.newBuilder().setMessage(Any.pack(cheese3)).build();
+
+        //Act
+        itemMarket.updateItem(build);
+        itemMarket.updateItem(build2);
+        itemMarket.updateItem(build3);
+
+        //Assert
+        assertEquals(2, allItems.get(CHEESE_ITEM_NAME).size());
+
+        assertEquals(6, productsQuantity.get(CHEESE_ITEM_NAME));
+        assertEquals(1, productsQuantity.size());
+    }
+    @Test
+    public void updateItemThrowsExceptionWhenPassingRequestOfDifferentType() {
+        //Arrange
+        itemMarket.addTrackedItem("cheese");
+        TickResponse cheese = TickResponse.newBuilder().setGoodName("cheese").setQuantity(2).setPrice(2.6).setOrigin(Origin.ASIA).build();
+        Aurora.AuroraResponse build = Aurora.AuroraResponse.newBuilder().setMessage(Any.pack(cheese)).build();
+
+
+        //Act
+        itemMarket.updateItem(build);
```

Comment:
why aren't we asserting or expecting the exception?

Reply:
Fixed